sale_coupon_advanced/models/sale_coupon.py

```diff
@@ -0,0 +1,38 @@
+import secrets
+import string
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class SaleCoupon(models.Model):
+    """Extend to implement custom code generation rule."""
+
+    _inherit = "sale.coupon"
+
+    # Must overwrite here for override method to take effect.
+    code = fields.Char(default=lambda s: s._generate_code())
```

Comment:
you could add two new fields in a program and implements in `_generate_code` if they are present in the program, you bypass the code generation. With an option as this sale_coupon module does (quite badly although but defined in a program specifically is more strict than adding a custom value in a wizard).
If it was a client specific module i would say nothing, but this is meant to be merged in OCA ? right ?